Dockerfile

```diff
@@ -18,22 +21,32 @@ FROM eclipse-temurin:17-jdk-alpine
 RUN apk update && apk upgrade --no-cache libcrypto3 libssl3
 # fix CVE-2023-52425
 RUN apk upgrade --no-cache libexpat
+RUN apk add --no-cache su-exec
 
 ENV OTEL_TRACES_EXPORTER="none"
 ENV OTEL_METRICS_EXPORTER="none"
 ENV OTEL_LOGS_EXPORTER="none"
 
+# Local storage dir configured in the default aidial.settings.json
+ENV STORAGE_DIR /app/data
+ENV LOG_DIR /app/log
+
 WORKDIR /app
 
 RUN adduser -u 1001 --disabled-password --gecos "" appuser
 
 COPY --from=builder --chown=appuser:appuser /build/ .
-RUN mkdir /app/log && chown -R appuser:appuser /app
+RUN chown -R appuser:appuser /app
 
-USER appuser
+COPY --chown=appuser:appuser docker-entrypoint.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/docker-entrypoint.sh
 
 HEALTHCHECK --start-period=30s --interval=1m --timeout=3s \
   CMD wget --no-verbose --spider --tries=1 http://localhost:8080/health || exit 1
 
 EXPOSE 8080 9464
-ENTRYPOINT ["/app/bin/aidial-core"]
+
+RUN mkdir -p "$LOG_DIR" && chown -R appuser:appuser "$LOG_DIR"
```

Comment:
can we combine RUN commands in one. I also think it’s worth raising them higher that HEALTHCHECK and EXPOSE

Reply:
fixed